routes/_middleware.ts

```diff
@@ -15,6 +29,36 @@ export const handler = async (
 
   const begin = performance.now();
   const url = new URL(req.url);
+  // FIXME (mcandeia) compatibility only.
+  if (
+    url.searchParams.has("editorData") &&
+    !url.pathname.startsWith("/live/editorData")
+  ) {
+    url.pathname = "/live/editorData";
+    return redirectTo(url);
+  }
+
+  if (
+    url.pathname.startsWith("/_live/workbench") &&
+    !url.searchParams.has("key") && !url.searchParams.has("pageId")
+  ) {
+    url.pathname = "/live/workbench";
+    return redirectTo(url);
+  }
+
+  if (
+    !url.pathname.startsWith("/live/previews") && url.searchParams.has("key") &&
+    !url.searchParams.has("editorData")
+  ) {
+    return previewSection(url, url.searchParams.get("key")!);
+  }
+  if (
+    !url.pathname.startsWith("/live/previews") &&
+    url.searchParams.has("pageId") &&
+    !url.searchParams.has("editorData")
```

Comment:
redirects to editor data and preview page depending on the specified query params